src/openrct2/network/NetworkBase.cpp

```diff
@@ -1030,41 +1029,40 @@ void NetworkBase::LoadGroups()
     platform_get_user_directory(path, nullptr, sizeof(path));
     safe_strcat_path(path, "groups.json", sizeof(path));
 
-    json_t* json = nullptr;
+    json_t jsonGroupConfig;
     if (Platform::FileExists(path))
     {
         try
         {
-            json = Json::ReadFromFile(path);
+            jsonGroupConfig = Json::ReadFromFile(path);
         }
         catch (const std::exception& e)
         {
             log_error("Failed to read %s as JSON. Setting default groups. %s", path, e.what());
         }
     }
 
-    if (json == nullptr)
+    if (!jsonGroupConfig.is_object())
     {
         SetupDefaultGroups();
     }
     else
     {
-        json_t* json_groups = json_object_get(json, "groups");
-        size_t groupCount = json_array_size(json_groups);
-        for (size_t i = 0; i < groupCount; i++)
+        json_t jsonGroups = jsonGroupConfig["groups"];
+        if (jsonGroups.is_array())
         {
-            json_t* jsonGroup = json_array_get(json_groups, i);
-
-            auto newgroup = std::make_unique<NetworkGroup>(NetworkGroup::FromJson(jsonGroup));
-            group_list.push_back(std::move(newgroup));
+            for (auto& jsonGroup : jsonGroups)
```

Comment:
This one doesn't work because NetworkGroup::FromJson doesn't accept a const jsonGroup

Reply:
Could the signature of `NetworkGroup::FromJson` be modified to accept it?

Reply:
(Also brought up [here](https://github.com/OpenRCT2/OpenRCT2/pull/12667#issuecomment-673454586))

Not without extensive modification of this PR. A const json_t will throw an exception if you try to access a member that doesn't exist (i.e. `json["key"]` where "key" doesn't exist) whereas a non-const json_t will just return a null json_t object. Changing parameters to const would require that this:

```
std::string someString = Json::GetString(json["somestring"], "default value");
```
be changed to:
```
std::string someString = "default value";
if ( json.contains("somestring") )
{
    someString = Json::GetString(json["somestring"], "default value");
}
```
which is a little verbose, especially considering this is done 10-20 lines in a row in places.

Alternatively, we could define a Json::GetObject method:
```
std::string someString = Json::GetString(Json::GetObject(json, "somestring"), "default value");
```
but, having had to understand all the existing code where something similar is done, this is nowhere near as clear as the current implementation.

So not having const parameters is a compromise for code readability. As I mentioned [here](https://github.com/OpenRCT2/OpenRCT2/pull/12667#issuecomment-687581380), I do intend to add comments where const would otherwise be defined to make this clear. (@tupaschoal I'd forgotten that you'd mentioned to ignore these)